### DISCUSSION TITLE
Potential issue with multiple releases of Pepys in use

### DISCUSSION THREAD

**Discussion by IanMayo** [debrief/pepys-import#787] [Development]
Hello @robintw & @BarisSari ,
I've about to send a new Beta into the clients, for them to test the new elements of Releases 2 and 3.

My plan is to get them to unpack the distribution into a dated folder. I'll then ask the Admin to run `install_pepys.bat`, then open Pepys Admin and run the database migration.

The current Pepys and this Beta version are looking at the same Postgres instance - but the migration may mean the database isn't now in the state that the non-Beta version expects.

The "current" version they have is 2021/01/15.  Looking at the Alembic migrations, I don't think the two since then would cause any issues:
* [allow the Tasks.ParentId field to be nullable](https://github.com/debrief/pepys-import/blob/develop/migrations/postgres_versions/2021-01-21_1be821cb7908_make_parent_id_nullable.py)
* [allow longer DataFile reference strings](https://github.com/debrief/pepys-import/blob/develop/migrations/postgres_versions/2021-02-23_80bc8b0d199b_increase_identifier.py)

**Comment by robintw** [comment]
Hopefully it won't cause any issues, but I think @BarisSari is the expert in Alembic/SQLAlchemy. There's a possibility it might raise an error if it finds the database doesn't match the defined schema - I don't know if it would raise an error if the nullability or length was different.

(This will definitely be a problem when we release https://github.com/debrief/pepys-import/pull/783, as I know a difference in field name will cause an error).

What do you think @BarisSari?

**Comment by BarisSari** [comment]
Hello @IanMayo, and @robintw. I think the Tasks table is still not used. So, I don't think nullability will cause an error. Actually, it was not possible to create an entity before this migration. I think it also shows that the table is empty :)

We increased `identifier` field from 10 chars to 30 chars. So, entities will satisfy the length constraint. I don't think it will cause an error, either.

**Comment by IanMayo** [comment]
Great - thanks Guys.
